File: ListPrimeFactors.cs
using System;
using System.Collections.Generic;

namespace Arithmetic_CS
{
    class ListPrimeFactors
    {
        private List<int> found_prime_factors;
        private int find_my_factors;
        private int index;

        public ListPrimeFactors(int val)
        {
            found_prime_factors = new List<int>();
            find_my_factors = val;
            // STEP 1:
            index = 2;
        }

        /**
        * Our function checks 'find_my_factors';
        * If it finds a factor, it records this factor,
        * then divides 'find_my_factors' by the found factor
        * and makes this the new 'find_my_factors'.
        * It continues recursively until all factors are found.
        * @param  - integer.
         */
        private int onlyPrimeFactors(int in_question)
        {
            int temp_limit;
            temp_limit = (int)Math.Ceiling(Math.Sqrt(in_question));

            while (index <= temp_limit)
            {
                // STEP 4:
                if (index != 1 && (in_question % index) == 0)
                { // avoid an infinite loop with the i != 1 check.
                    found_prime_factors.Add(index);
                    // STEP 2, 3:
                    return onlyPrimeFactors(in_question / index);
                }
                index++;
            }
            found_prime_factors.Add(in_question);

            return 0;
        }

        /**
         * Renders the prime factors to screen.
         * @return - a list
         */
        public List<int> getPrimeFactors()
        {
            onlyPrimeFactors(find_my_factors);
            return found_prime_factors;
        }
    }
}

File: ListPrimeFactors.cpp
#include "stdafx.h"
#include "ListPrimeFactors.h"


ListPrimeFactors::ListPrimeFactors(unsigned val) {
	found_prime_factors = {};
	find_my_factors = val;
	aux << val;
	render_factors = "The prime factors of " + aux.str() + " are: \n";
	i = 2;
}

int ListPrimeFactors::onlyPrimeFactors(int in_question) {
	int temp_limit;
	temp_limit = (int)ceil(sqrt(in_question));

	while (i <= temp_limit) {
		if (i != 1 && (in_question % i) == 0) { // avoid an infinite loop with the i != 1 check.
			found_prime_factors.push_back(i);
			return onlyPrimeFactors(in_question / i);
		}
		i++;
	}
	found_prime_factors.push_back(in_question);

	return 0;
}

/**
* Renders the prime prime factors to screen.
*/
string ListPrimeFactors::getPrimeFactors() {
	onlyPrimeFactors(find_my_factors);
	found_prime_factors.shrink_to_fit();

	//iterate through and retrieve members
	for (unsigned factor : found_prime_factors) {
		aux.str("");
		aux << factor;
		render_factors += aux.str() + " X ";
	}
	render_factors = render_factors.substr(0, render_factors.length() - 3);

	return render_factors;
}


ListPrimeFactors::~ListPrimeFactors() {
}
